### PR TITLE
Misc updates for tracing and packaging

### PR DESCRIPTION
**Pull request by cgillum**
- Added more detail to dequeue tracing
- Fix nuget warnings in package generation
- Upgrade to non-preview version of Microsoft.SourceLink.GitHub
- Minor code cleanup

### CODE REVIEW

File: src/DurableTask.AzureStorage/DurableTask.AzureStorage.csproj [hunk 1]
```diff
@@ -19,7 +19,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0-*" PrivateAssets="All" />
+    <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.*" PrivateAssets="All" />
```
Comment (ConnorMcMahon) [inline src/DurableTask.AzureStorage/DurableTask.AzureStorage.csproj]: Any specific reason for this change in versioning?

File: src/DurableTask.Core/DurableTask.Core.csproj [hunk 2]
```diff
@@ -5,6 +5,7 @@
     <TargetFrameworks>netstandard2.0;net451</TargetFrameworks>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <PackageId>Microsoft.Azure.DurableTask.Core</PackageId>
+    <GeneratePackageOnBuild>false</GeneratePackageOnBuild>
```
Comment (ConnorMcMahon) [inline src/DurableTask.Core/DurableTask.Core.csproj]: Is this something we want to commit?

File: src/DurableTask.AzureStorage/Messaging/OrchestrationSession.cs [hunk 3]
```diff
@@ -96,6 +96,8 @@ public void AddOrReplaceMessages(IEnumerable<MessageData> messages)
                 return null; // timed-out
             }
 
+            this.StartNewLogicalTraceScope();
```
Comment (ConnorMcMahon) [inline src/DurableTask.AzureStorage/Messaging/OrchestrationSession.cs]: This is just to make sure we aren't missing activity ids for logs here?

Comment (cgillum) [reply src/DurableTask.AzureStorage/Messaging/OrchestrationSession.cs]: Right. When I was debugging that performance issue, I noticed we were missing a bunch of Activity IDs for the extended sessions case. This line of code fixes that.

File: src/DurableTask.Core/DurableTask.Core.csproj (hunk 2 above)
Comment (cgillum) [reply src/DurableTask.Core/DurableTask.Core.csproj]: Not sure, I kind of wish I'd left that out.

File: src/DurableTask.AzureStorage/DurableTask.AzureStorage.csproj (hunk 1 above)
Comment (cgillum) [reply src/DurableTask.AzureStorage/DurableTask.AzureStorage.csproj]: `1.0.0-*` mapped to preview versions, like `1.0.0-beta2-19554-01`. There is a `1.0.0` version out now so I wanted to update our reference to get that build.

**Pull request merged**